Day1/Shubham_Nagaria_Next_Permutation.cpp

```diff
@@ -0,0 +1,34 @@
+#Problem Statement - https://leetcode.com/problems/next-permutation/
+class Solution {
+public:
+    void nextPermutation( vector<int>& nums ) {
+        int lenOfnums = nums.size(), indexOfFirstSwap, indexOfnextSwap, indexForNextgreaterElement, diff, maxDiff = INT_MAX;
+        // finding the first index where element can swap with. (lets take as indexOfFirstSwap)
```

Comment:
This problem clearly has a few independent steps. Individual steps can be written in separate methods and the `nextPermutation` will read like what it is doing. Forcing to read the entire low level implementation of algo just to understand what you're doing is not fair.